app/app.gradle.kts

```diff
@@ -31,6 +31,9 @@ dependencies {
     compile(group = "de.huxhorn.lilith", name = "de.huxhorn.lilith.3rdparty.junique", version = "1.0.4")
     compile(group = "org.apache.commons", name = "commons-csv", version = "1.5")
     testCompile(project("test_plugins"))
+    val apiTestOutput = project(":api").dependencyProject.sourceSets["test"].output
```

Comment:
I think you can just depend upon the output of the `test` configuration for the `API` project?
Otherwise, I generally recommend creating a separate project that contains test logic you want to share.

You're allowed to have a dependency like this (-> means depends on):

project A:Test -> project ATestUtil -> project A
project A:Test -> project A

Where project A:Test is the test code for project A

Reply:
Mmm, that's possible. I just grabbed this configuration logic from GRIP

Reply:
What I've suggested is cleaner.
I don't know if the task graph will generate correctly with what you have here.
It might work in most cases due to timing, but no guarantees. Maybe I'm wrong.

You'd have to look at the `compileTest` task graph with `--dry-run`.